README.rst
Paco
=========================

Paco is a project that will hold a couple of apps I'm developing for my company.


HISTORY.rst
.. :changelog:

History
-------

0.1 (2012-12-19)
+++++++++++++++++++






paco/wsgi.py
"""
WSGI config for paco project.

This module contains the WSGI application used by Django's development server
and any production WSGI deployments. It should expose a module-level variable
named ``application``. Django's ``runserver`` and ``runfcgi`` commands discover
this application via the ``WSGI_APPLICATION`` setting.

Usually you will have the standard Django WSGI application here, but it also
might make sense to replace the whole Django WSGI application with a custom one
that later delegates to the Django one. For example, you could introduce WSGI
middleware here, or combine a Django application with an application of another
framework.

"""
import os

os.environ.setdefault("DJANGO_SETTINGS_MODULE", "paco.settings")

# This application object is used by any WSGI server configured to use this
# file. This includes Django's development server, if the WSGI_APPLICATION
# setting points here.
from django.core.wsgi import get_wsgi_application
application = get_wsgi_application()

# Apply WSGI middleware here.
# from helloworld.wsgi import HelloWorldApplication
# application = HelloWorldApplication(application)


paco/settings.py
# Django settings for paco project.
import os

PROJECT_ROOT = os.path.dirname(__file__)

DEBUG = True
TEMPLATE_DEBUG = DEBUG

ADMINS = (
    # ('Your Name', 'your_email@example.com'),
)

MANAGERS = ADMINS

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3', # Add 'postgresql_psycopg2', 'mysql', 'sqlite3' or 'oracle'.
        'NAME': 'paco.db',                # Or path to database file if using sqlite3.
    }
}

# Local time zone for this installation. Choices can be found here:
# http://en.wikipedia.org/wiki/List_of_tz_zones_by_name
# although not all choices may be available on all operating systems.
# In a Windows environment this must be set to your system time zone.
TIME_ZONE = 'Europe/Amsterdam'

# Language code for this installation. All choices can be found here:
# http://www.i18nguy.com/unicode/language-identifiers.html
LANGUAGE_CODE = 'en-us'

SITE_ID = 1

# If you set this to False, Django will make some optimizations so as not
# to load the internationalization machinery.
USE_I18N = True

# If you set this to False, Django will not format dates, numbers and
# calendars according to the current locale.
USE_L10N = True

# If you set this to False, Django will not use timezone-aware datetimes.
USE_TZ = True

# Absolute filesystem path to the directory that will hold user-uploaded files.
# Example: "/home/media/media.lawrence.com/media/"
MEDIA_ROOT = os.path.join(PROJECT_ROOT, 'media')

# URL that handles the media served from MEDIA_ROOT. Make sure to use a
# trailing slash.
# Examples: "http://media.lawrence.com/media/", "http://example.com/media/"
MEDIA_URL = '/media/'

# Absolute path to the directory static files should be collected to.
# Don't put anything in this directory yourself; store your static files
# in apps' "static/" subdirectories and in STATICFILES_DIRS.
# Example: "/home/media/media.lawrence.com/static/"
STATIC_ROOT = os.path.join(PROJECT_ROOT, 'static')

# URL prefix for static files.
# Example: "http://media.lawrence.com/static/"
STATIC_URL = '/static/'

# Additional locations of static files
STATICFILES_DIRS = (
    # Put strings here, like "/home/html/static" or "C:/www/django/static".
    # Always use forward slashes, even on Windows.
    # Don't forget to use absolute paths, not relative paths.
)

# List of finder classes that know how to find static files in
# various locations.
STATICFILES_FINDERS = (
    'django.contrib.staticfiles.finders.FileSystemFinder',
    'django.contrib.staticfiles.finders.AppDirectoriesFinder',
#    'django.contrib.staticfiles.finders.DefaultStorageFinder',
)

# Make this unique, and don't share it with anybody.
SECRET_KEY = 'p=t9fho+47h1mbesyefiajok23r@93!au&amp;=uh8!&amp;e==_$u(1t1'

# List of callables that know how to import templates from various sources.
TEMPLATE_LOADERS = (
    'django.template.loaders.filesystem.Loader',
    'django.template.loaders.app_directories.Loader',
#     'django.template.loaders.eggs.Loader',
)

MIDDLEWARE_CLASSES = (
    'django.middleware.common.CommonMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    # Uncomment the next line for simple clickjacking protection:
    # 'django.middleware.clickjacking.XFrameOptionsMiddleware',
)

ROOT_URLCONF = 'paco.urls'

# Python dotted path to the WSGI application used by Django's runserver.
WSGI_APPLICATION = 'paco.wsgi.application'

TEMPLATE_DIRS = (
    # Put strings here, like "/home/html/django_templates" or "C:/www/django/templates".
    # Always use forward slashes, even on Windows.
    # Don't forget to use absolute paths, not relative paths.
)

INSTALLED_APPS = (
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.sites',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    # Uncomment the next line to enable the admin:
    'django.contrib.admin',
    # Uncomment the next line to enable admin documentation:
    'django.contrib.admindocs',
    'south',
    'apps.dees',
)

# A sample logging configuration. The only tangible logging
# performed by this configuration is to send an email to
# the site admins on every HTTP 500 error when DEBUG=False.
# See http://docs.djangoproject.com/en/dev/topics/logging for
# more details on how to customize your logging configuration.
LOGGING = {
    'version': 1,
    'disable_existing_loggers': False,
    'filters': {
        'require_debug_false': {
            '()': 'django.utils.log.RequireDebugFalse'
        }
    },
    'handlers': {
        'mail_admins': {
            'level': 'ERROR',
            'filters': ['require_debug_false'],
            'class': 'django.utils.log.AdminEmailHandler'
        }
    },
    'loggers': {
        'django.request': {
            'handlers': ['mail_admins'],
            'level': 'ERROR',
            'propagate': True,
        },
    }
}


apps/dees/models.py
from datetime import datetime

from django.db import models
from django.contrib.auth.models import User

class Project(models.Model):
	REASONS = (
				('Nieuwe klant', 'Nieuwe klant'), 
				('Bestaande klant', 'Bestaande klant'),
				('Interne opdracht', 'Interne opdracht'), 
				('Backfill', 'Backfill'),
			)

	title = models.CharField(max_length=255, 
							help_text="Please specify the title of the project")
	reason = models.CharField(max_length=16, choices=REASONS,
								help_text="Please specify the reason")
	location = models.CharField(max_length=128,
								help_text="Please specify the location")
	project_manager = models.ForeignKey(User, blank=True, null=True, on_delete=models.SET_NULL,
										related_name="project_mgr_of_project",
										help_text="Please specify the project manager")
	account_manager = models.ForeignKey(User, blank=True, null=True, on_delete=models.SET_NULL,
										related_name="account_mgr_of_project",
										help_text="Please specify the account manager")
	project_id = models.CharField(max_length=32,
							help_text="Please specify the SO or PID number")

	def __unicode__(self):
		return self.title


class Skill(models.Model):
	title = models.CharField(max_length=255,
							help_text="Please specify the name of the skill")
	description = models.TextField(help_text="Please specify a description of the skill")

	def __unicode__(self):
		return self.title


class Task(models.Model):
	WINDOWS = (
				('OH', 'Office Hours'),
				('EOH', 'Extended Office Hours'), 
				('OOH', 'Out of Office Hours'),
			)

	title = models.CharField(max_length=255,
							help_text="Please specify the title of the task")
	project = models.ForeignKey(Project)
	description = models.TextField(help_text="Please specify a description of the task that needs to be performed")
	prework = models.TextField(help_text="Please specify the prework that needs to be done")
	scope = models.TextField(help_text="Please specify the scope of work")
	skills = models.ManyToManyField(Skill)
	duration = models.FloatField(help_text="Please specify the number of hours this task will take")
	window = models.CharField(max_length=255, choices=WINDOWS,
								help_text="Please specify when this task needs to be performed")

	def __unicode__(self):
		return self.title

class ResourceRequest(models.Model):
	requester = models.ForeignKey(User, blank=True, null=True, on_delete=models.SET_NULL)
	project = models.ForeignKey(Project)
	task = models.ForeignKey(Task)
	request_date = models.DateTimeField(auto_now_add=True)
	start_date = models.DateTimeField('start date',
								help_text="Please specify the start date")
	end_date = models.DateTimeField('end date',
								help_text="Please specify the end date")

	def __unicode__(self):
		return self.id

apps/dees/migrations/0001_initial.py
# -*- coding: utf-8 -*-
import datetime
from south.db import db
from south.v2 import SchemaMigration
from django.db import models


class Migration(SchemaMigration):

    def forwards(self, orm):
        # Adding model 'Project'
        db.create_table('dees_project', (
            ('id', self.gf('django.db.models.fields.AutoField')(primary_key=True)),
            ('title', self.gf('django.db.models.fields.CharField')(max_length=255)),
            ('reason', self.gf('django.db.models.fields.CharField')(max_length=16)),
            ('location', self.gf('django.db.models.fields.CharField')(max_length=128)),
            ('project_manager', self.gf('django.db.models.fields.related.ForeignKey')(blank=True, related_name='project_mgr_of_project', null=True, on_delete=models.SET_NULL, to=orm['auth.User'])),
            ('account_manager', self.gf('django.db.models.fields.related.ForeignKey')(blank=True, related_name='account_mgr_of_project', null=True, on_delete=models.SET_NULL, to=orm['auth.User'])),
            ('project_id', self.gf('django.db.models.fields.CharField')(max_length=32)),
        ))
        db.send_create_signal('dees', ['Project'])

        # Adding model 'Skill'
        db.create_table('dees_skill', (
            ('id', self.gf('django.db.models.fields.AutoField')(primary_key=True)),
            ('title', self.gf('django.db.models.fields.CharField')(max_length=255)),
            ('description', self.gf('django.db.models.fields.TextField')()),
        ))
        db.send_create_signal('dees', ['Skill'])

        # Adding model 'Task'
        db.create_table('dees_task', (
            ('id', self.gf('django.db.models.fields.AutoField')(primary_key=True)),
            ('title', self.gf('django.db.models.fields.CharField')(max_length=255)),
            ('project', self.gf('django.db.models.fields.related.ForeignKey')(to=orm['dees.Project'])),
            ('description', self.gf('django.db.models.fields.TextField')()),
            ('prework', self.gf('django.db.models.fields.TextField')()),
            ('scope', self.gf('django.db.models.fields.TextField')()),
            ('duration', self.gf('django.db.models.fields.FloatField')()),
            ('window', self.gf('django.db.models.fields.CharField')(max_length=255)),
        ))
        db.send_create_signal('dees', ['Task'])

        # Adding M2M table for field skills on 'Task'
        db.create_table('dees_task_skills', (
            ('id', models.AutoField(verbose_name='ID', primary_key=True, auto_created=True)),
            ('task', models.ForeignKey(orm['dees.task'], null=False)),
            ('skill', models.ForeignKey(orm['dees.skill'], null=False))
        ))
        db.create_unique('dees_task_skills', ['task_id', 'skill_id'])

        # Adding model 'ResourceRequest'
        db.create_table('dees_resourcerequest', (
            ('id', self.gf('django.db.models.fields.AutoField')(primary_key=True)),
            ('requester', self.gf('django.db.models.fields.related.ForeignKey')(to=orm['auth.User'], null=True, on_delete=models.SET_NULL, blank=True)),
            ('project', self.gf('django.db.models.fields.related.ForeignKey')(to=orm['dees.Project'])),
            ('task', self.gf('django.db.models.fields.related.ForeignKey')(to=orm['dees.Task'])),
            ('request_date', self.gf('django.db.models.fields.DateTimeField')()),
            ('start_date', self.gf('django.db.models.fields.DateTimeField')()),
            ('end_date', self.gf('django.db.models.fields.DateTimeField')()),
        ))
        db.send_create_signal('dees', ['ResourceRequest'])


    def backwards(self, orm):
        # Deleting model 'Project'
        db.delete_table('dees_project')

        # Deleting model 'Skill'
        db.delete_table('dees_skill')

        # Deleting model 'Task'
        db.delete_table('dees_task')

        # Removing M2M table for field skills on 'Task'
        db.delete_table('dees_task_skills')

        # Deleting model 'ResourceRequest'
        db.delete_table('dees_resourcerequest')


    models = {
        'auth.group': {
            'Meta': {'object_name': 'Group'},
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'name': ('django.db.models.fields.CharField', [], {'unique': 'True', 'max_length': '80'}),
            'permissions': ('django.db.models.fields.related.ManyToManyField', [], {'to': "orm['auth.Permission']", 'symmetrical': 'False', 'blank': 'True'})
        },
        'auth.permission': {
            'Meta': {'ordering': "('content_type__app_label', 'content_type__model', 'codename')", 'unique_together': "(('content_type', 'codename'),)", 'object_name': 'Permission'},
            'codename': ('django.db.models.fields.CharField', [], {'max_length': '100'}),
            'content_type': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['contenttypes.ContentType']"}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'name': ('django.db.models.fields.CharField', [], {'max_length': '50'})
        },
        'auth.user': {
            'Meta': {'object_name': 'User'},
            'date_joined': ('django.db.models.fields.DateTimeField', [], {'default': 'datetime.datetime.now'}),
            'email': ('django.db.models.fields.EmailField', [], {'max_length': '75', 'blank': 'True'}),
            'first_name': ('django.db.models.fields.CharField', [], {'max_length': '30', 'blank': 'True'}),
            'groups': ('django.db.models.fields.related.ManyToManyField', [], {'to': "orm['auth.Group']", 'symmetrical': 'False', 'blank': 'True'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'is_active': ('django.db.models.fields.BooleanField', [], {'default': 'True'}),
            'is_staff': ('django.db.models.fields.BooleanField', [], {'default': 'False'}),
            'is_superuser': ('django.db.models.fields.BooleanField', [], {'default': 'False'}),
            'last_login': ('django.db.models.fields.DateTimeField', [], {'default': 'datetime.datetime.now'}),
            'last_name': ('django.db.models.fields.CharField', [], {'max_length': '30', 'blank': 'True'}),
            'password': ('django.db.models.fields.CharField', [], {'max_length': '128'}),
            'user_permissions': ('django.db.models.fields.related.ManyToManyField', [], {'to': "orm['auth.Permission']", 'symmetrical': 'False', 'blank': 'True'}),
            'username': ('django.db.models.fields.CharField', [], {'unique': 'True', 'max_length': '30'})
        },
        'contenttypes.contenttype': {
            'Meta': {'ordering': "('name',)", 'unique_together': "(('app_label', 'model'),)", 'object_name': 'ContentType', 'db_table': "'django_content_type'"},
            'app_label': ('django.db.models.fields.CharField', [], {'max_length': '100'}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'model': ('django.db.models.fields.CharField', [], {'max_length': '100'}),
            'name': ('django.db.models.fields.CharField', [], {'max_length': '100'})
        },
        'dees.project': {
            'Meta': {'object_name': 'Project'},
            'account_manager': ('django.db.models.fields.related.ForeignKey', [], {'blank': 'True', 'related_name': "'account_mgr_of_project'", 'null': 'True', 'on_delete': 'models.SET_NULL', 'to': "orm['auth.User']"}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'location': ('django.db.models.fields.CharField', [], {'max_length': '128'}),
            'project_id': ('django.db.models.fields.CharField', [], {'max_length': '32'}),
            'project_manager': ('django.db.models.fields.related.ForeignKey', [], {'blank': 'True', 'related_name': "'project_mgr_of_project'", 'null': 'True', 'on_delete': 'models.SET_NULL', 'to': "orm['auth.User']"}),
            'reason': ('django.db.models.fields.CharField', [], {'max_length': '16'}),
            'title': ('django.db.models.fields.CharField', [], {'max_length': '255'})
        },
        'dees.resourcerequest': {
            'Meta': {'object_name': 'ResourceRequest'},
            'end_date': ('django.db.models.fields.DateTimeField', [], {}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'project': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['dees.Project']"}),
            'request_date': ('django.db.models.fields.DateTimeField', [], {}),
            'requester': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['auth.User']", 'null': 'True', 'on_delete': 'models.SET_NULL', 'blank': 'True'}),
            'start_date': ('django.db.models.fields.DateTimeField', [], {}),
            'task': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['dees.Task']"})
        },
        'dees.skill': {
            'Meta': {'object_name': 'Skill'},
            'description': ('django.db.models.fields.TextField', [], {}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'title': ('django.db.models.fields.CharField', [], {'max_length': '255'})
        },
        'dees.task': {
            'Meta': {'object_name': 'Task'},
            'description': ('django.db.models.fields.TextField', [], {}),
            'duration': ('django.db.models.fields.FloatField', [], {}),
            'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
            'prework': ('django.db.models.fields.TextField', [], {}),
            'project': ('django.db.models.fields.related.ForeignKey', [], {'to': "orm['dees.Project']"}),
            'scope': ('django.db.models.fields.TextField', [], {}),
            'skills': ('django.db.models.fields.related.ManyToManyField', [], {'to': "orm['dees.Skill']", 'symmetrical': 'False'}),
            'title': ('django.db.models.fields.CharField', [], {'max_length': '255'}),
            'window': ('django.db.models.fields.CharField', [], {'max_length': '255'})
        }
    }

    complete_apps = ['dees']